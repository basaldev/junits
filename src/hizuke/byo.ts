export default function byo(date: Date, unit: string = 'ç§’'): string {
  return `${date.getSeconds()}${unit}`;
}
