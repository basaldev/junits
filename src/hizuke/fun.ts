export default function fun(date: Date, unit: string = 'åˆ†'): string {
  return `${date.getMinutes()}${unit}`;
}
