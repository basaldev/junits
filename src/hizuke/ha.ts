import converToDate, { Datetime } from './convertToDate';
import yoriMae from './yoriMae';
import yoriAto from './yoriAto';
import toOnaji from './toOnaji';
import toOnajiKaMae from './toOnajiKaMae';
import toOnajiKaAto from './toOnajiKaAto';
import noAida from './noAida';

export default function ha(date1: Date, date2: Date) {
  return {
    yoriMae: () => yoriMae(date1, date2),
    yoriAto: () => yoriAto(date1, date2),
    toOnaji: () => toOnaji(date1, date2),
    toOnajiKaMae: () => toOnajiKaMae(date1, date2),
    toOnajiKaAto: () => toOnajiKaAto(date1, date2),
    to: (date3: Datetime) => ({ noAida: () => noAida(date1, date2, converToDate(date3)) })
  };
}
