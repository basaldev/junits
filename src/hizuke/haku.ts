export default function haku(char: string = ' '): string {
  return char;
}
