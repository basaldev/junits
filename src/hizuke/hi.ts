export default function hi(date: Date, unit: string = 'æ—¥'): string {
  return `${date.getDate()}${unit}`;
}
