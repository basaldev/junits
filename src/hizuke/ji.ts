export default function ji(date: Date, unit: string = 'æ™‚'): string {
  return `${date.getHours()}${unit}`;
}
