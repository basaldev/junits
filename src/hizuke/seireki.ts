export default function seireki(date: Date, unit: string = 'å¹´'): string {
  return `${date.getFullYear()}${unit}`;
}
