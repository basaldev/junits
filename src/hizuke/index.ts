import seireki from './seireki';
import wareki from './wareki';
import tsuki from './tsuki';
import hi from './hi';
import tanYobi from './tanYobi';
import yobi from './yobi';
import ji from './ji';
import fun from './fun';
import byo from './byo';
import haku from './haku';
import ha from './ha';

export default function hizuke(date: Date = new Date(), state = []) {
  function setState(value) {
    state.push(value);
    return hizuke(date, state);
  }

  function fixDigit(value) {
    const index = state.length - 1;
    state[index] = state[index].replace(/^(.*?)(\d+)(.*?)$/, (...args) =>
      args[1] + `00000${args[2]}`.slice(-value) + args[3]);
    return hizuke(date, state);
  }
  
  function output() {
    return state.join('');
  }

  return {
    seireki: (unit: string) => setState(seireki(date, unit)),
    wareki: (unit: string) => setState(wareki(date, 'full', unit)),
    tanWareki: (unit: string) => setState(wareki(date, 'short', unit)),
    alphaWareki: (unit: string) => setState(wareki(date, 'alpha', unit)),
    tsuki: (unit: string) => setState(tsuki(date, unit)),
    hi: (unit: string) => setState(hi(date, unit)),
    tanYobi: (unit: string) => setState(tanYobi(date, unit)),
    yobi: (unit: string) => setState(yobi(date, unit)),
    ji: (unit: string) => setState(ji(date, unit)),
    fun: (unit: string) => setState(fun(date, unit)),
    byo: (unit: string) => setState(byo(date, unit)),
    haku:(value: string) =>  setState(haku(value)),
    niketa: () => fixDigit(2),
    ijo: output,
    ha:(date2: Date) => ha(date, date2),
  }
}