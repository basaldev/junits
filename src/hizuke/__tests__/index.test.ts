jest.mock('../seireki');
jest.mock('../wareki');
jest.mock('../tsuki');
jest.mock('../hi');
jest.mock('../tanYobi');
jest.mock('../yobi');
jest.mock('../ji');
jest.mock('../fun');
jest.mock('../byo');
jest.mock('../haku');
jest.mock('../ha');

import seireki from '../seireki';
import wareki from '../wareki';
import tsuki from '../tsuki';
import hi from '../hi';
import tanYobi from '../tanYobi';
import yobi from '../yobi';
import ji from '../ji';
import fun from '../fun';
import byo from '../byo';
import haku from '../haku';
import ha from '../ha';
import hizuke from '../index';

describe('hizuke', () => {
  let date0;
  let date1;
  let date2;
  let expectedHizukeReturn;
  let originalDate;

  beforeAll(() => {
    date0 = new Date(2019, 4, 7, 13, 16, 20);
    originalDate = Date;
    // @ts-ignore
    Date = class extends originalDate {
      constructor(...args) {
        super();
        if (args.length) {
          return new originalDate(...args);
        }
        return date0;
      }
    };
  });

  afterAll(() => {
    Date = originalDate;
  });

  beforeEach(() => {
    date1 = new Date(2019, 4, 7, 13, 16, 20);
    date2 = new Date(2020, 4, 7, 13, 16, 20);
    expectedHizukeReturn = {
      seireki: expect.any(Function),
      wareki: expect.any(Function),
      tanWareki: expect.any(Function),
      alphaWareki: expect.any(Function),
      tsuki: expect.any(Function),
      hi: expect.any(Function),
      tanYobi: expect.any(Function),
      yobi: expect.any(Function),
      ji: expect.any(Function),
      fun: expect.any(Function),
      byo: expect.any(Function),
      haku: expect.any(Function),
      niketa: expect.any(Function),
      moji: expect.any(Function),
      atai: expect.any(Function),
      ha: expect.any(Function)
    };
  });

  it('should return proper members', () => {
    expect(hizuke(date1)).toMatchSnapshot(expectedHizukeReturn);
  });

  it('should create new date instance without the first arg', () => {
    hizuke().seireki('unit');
    expect(seireki).toHaveBeenLastCalledWith(date0, 'unit');
  });

  it('should call seireki', () => {
    expect(hizuke(date1).seireki('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(seireki).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call wareki', () => {
    expect(hizuke(date1).wareki('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(wareki).toHaveBeenLastCalledWith(date1, 'full', 'unit');
  });

  it('should call tanWareki', () => {
    expect(hizuke(date1).tanWareki('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(wareki).toHaveBeenLastCalledWith(date1, 'short', 'unit');
  });

  it('should call alphaWareki', () => {
    expect(hizuke(date1).alphaWareki('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(wareki).toHaveBeenLastCalledWith(date1, 'alpha', 'unit');
  });

  it('should call tsuki', () => {
    expect(hizuke(date1).tsuki('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(tsuki).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call hi', () => {
    expect(hizuke(date1).hi('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(hi).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call tanYobi', () => {
    expect(hizuke(date1).tanYobi('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(tanYobi).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call yobi', () => {
    expect(hizuke(date1).yobi('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(yobi).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call ji', () => {
    expect(hizuke(date1).ji('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(ji).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call fun', () => {
    expect(hizuke(date1).fun('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(fun).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call byo', () => {
    expect(hizuke(date1).byo('unit')).toMatchSnapshot(expectedHizukeReturn);
    expect(byo).toHaveBeenLastCalledWith(date1, 'unit');
  });

  it('should call haku', () => {
    expect(hizuke(date1).haku('space')).toMatchSnapshot(expectedHizukeReturn);
    expect(haku).toHaveBeenLastCalledWith('space');
  });

  it('should call niketa/fixDigit', () => {
    const myHizuke2 = jest.fn(hizuke);
    expect(myHizuke2(date1, ['令和2年']).niketa()).toMatchSnapshot(expectedHizukeReturn);
    expect(myHizuke2).toHaveBeenLastCalledWith(date1, ['令和02年']);
  });

  it('should call moji/output', () => {
    expect(hizuke(date1, ['1月', '2日']).moji()).toBe('1月2日');
  });

  it('should call atai/value', () => {
    expect(hizuke(date1).atai()).toEqual({
      byo: 20,
      fun: 16,
      hi: 7,
      ji: 13,
      seireki: 2019,
      tsuki: 5,
      wareki: {
        alphaGengo: 'R',
        gengo: '令和',
        nen: 1,
        tanGengo: '令'
      }
    });
  });

  it('should call ha', () => {
    hizuke(date1).ha(date2);
    expect(ha).toHaveBeenLastCalledWith(date1, date2);
  });
});
