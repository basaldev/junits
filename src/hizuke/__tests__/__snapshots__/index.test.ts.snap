// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hizuke should call alphaWareki 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call byo 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call fun 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call haku 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call hi 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call ji 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call niketa/fixDigit 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call seireki 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call tanWareki 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call tanYobi 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call tsuki 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call wareki 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should call yobi 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;

exports[`hizuke should return proper members 1`] = `
Object {
  "alphaWareki": Any<Function>,
  "byo": Any<Function>,
  "fun": Any<Function>,
  "ha": Any<Function>,
  "haku": Any<Function>,
  "hi": Any<Function>,
  "ji": Any<Function>,
  "moji": Any<Function>,
  "niketa": Any<Function>,
  "seireki": Any<Function>,
  "tanWareki": Any<Function>,
  "tanYobi": Any<Function>,
  "tsuki": Any<Function>,
  "wareki": Any<Function>,
  "yobi": Any<Function>,
}
`;
